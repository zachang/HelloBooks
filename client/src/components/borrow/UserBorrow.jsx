import React from 'react';
import PropTypes from 'react-proptypes';

const UserBorrow = props => (
    <div className='col l4 s10 m6 cardsm'>
      <div className='card large sticky-action'>
        <div className='card-image waves-effect waves-block waves-light'>
          <img className='activator'
               src={(props.borrow.Book.book_image !== null) ? props.borrow.Book.book_image : '../../imgs/default.jpg'}/>
        </div>
        <div className='card-content'>
        <span className='card-title activator grey-text text-darken-4'>
          <small>{ props.borrow.Book.book_name }</small>
          <i className='material-icons right'>more_vert</i>
        </span>
        </div>
        <div className='card-reveal'>
        <span className='card-title grey-text text-darken-4'>
         <small>{ props.borrow.Book.book_name }</small>
          <i className='material-icons right'>
            close
          </i>
        </span>
          <p>Author: { props.borrow.Book.author }</p>
          <p>Category: { props.borrow.Book.Category.category_name }</p>
          <p>Published: { new Date(props.borrow.Book.publish_year).getFullYear() }</p>
          <p>Pages: { props.borrow.Book.pages }</p>
          <p>Stock: { props.borrow.Book.book_count }</p>
          <p>Description: {
            (props.borrow.Book.description)
          }
          </p>
        </div>
        <div className='card-action home-card'>
          {(props.borrow.borrow_status === 'pending') ?
            <button
              className='waves-effect waves-light btn grey tooltipped'
              data-position='bottom'
              data-delay='50'
              data-tooltip='Wait for borrow confirmation'
            >
              Return
            </button> : (props.borrow.returned === 'true') ?
            <button
              className='waves-effect waves-light btn grey tooltipped'
              data-position='bottom'
              data-delay='50'
              data-tooltip='Book return confirmed'
            >
              Confirmed
            </button> :
            <button
              onClick={ () => (props.returnBook(props.borrow.Book.id )) }
              className='waves-effect waves-light btn red'
            >
              Return
            </button>}
        </div>
      </div>
    </div>
  );

UserBorrow.propTypes = {
  borrow: PropTypes.object.isRequired,

};
export default UserBorrow;
