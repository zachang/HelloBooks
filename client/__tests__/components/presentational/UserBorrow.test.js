import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';

import UserBorrow from '../../../src/components/borrow/UserBorrow.jsx';

describe('<UserBorrow/>', () => {
  const props = {
    borrow: {
      Book: {
        bookImage: 'ggv.jpg',
        Category: {
          categoryName: 'Arts'
        }
      }
    },
    returnBook: () => ''
  };

  const shallowWrapper = shallow(<UserBorrow {...props}/>);
  const tree = toJson(shallowWrapper);
  let wrapper;

  const wrapperMount = mount(<UserBorrow {...props} />);

  it('renders <UserBorrow /> component', () => {
    expect(tree).toMatchSnapshot();
  });

  it('calls onClick', () => {
    wrapper = shallow(<UserBorrow {...props} />);
    wrapper.find('.red').simulate('click');
    expect(wrapperMount.find('.red').at(0).text()).toBe('Return');
  });
});

