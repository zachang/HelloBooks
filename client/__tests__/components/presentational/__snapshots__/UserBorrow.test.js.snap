// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserBorrow/> renders <UserBorrow /> component 1`] = `
<div
  className="col l4 s10 m6 cardsm"
>
  <div
    className="card large sticky-action"
  >
    <div
      className="card-image waves-effect waves-block waves-light"
    >
      <img
        className="activator"
        src="ggv.jpg"
      />
    </div>
    <div
      className="card-content"
    >
      <span
        className="card-title activator grey-text text-darken-4"
      >
        <small />
        <i
          className="material-icons right"
        >
          more_vert
        </i>
      </span>
    </div>
    <div
      className="card-reveal"
    >
      <span
        className="card-title grey-text text-darken-4"
      >
        <small />
        <i
          className="material-icons right"
        >
          close
        </i>
      </span>
      <p>
        Author: 
      </p>
      <p>
        Category: 
        Arts
      </p>
      <p>
        Published: 
        NaN
      </p>
      <p>
        Pages: 
      </p>
      <p>
        Stock: 
      </p>
      <p>
        Description: 
      </p>
    </div>
    <div
      className="card-action home-card"
    >
      <button
        className="waves-effect waves-light btn red"
        onClick={[Function]}
      >
        Return
      </button>
    </div>
  </div>
</div>
`;
